app/liquid_tags/codesharer_tag.rb

```diff
@@ -0,0 +1,54 @@
+class CodesharerTag < LiquidTagBase
+  PARTIAL = "liquids/codesharer".freeze
+  PARAM_REGEXP = /\w|(#|=)/.freeze
+
+  def initialize(_tag_name, link, _parse_context)
+    super
+    stripped_link = ActionController::Base.helpers.strip_tags(link)
+    the_link = stripped_link.split(" ").first
+    @embedded_url = CodesharerTag.embedded_url(the_link)
```

Comment:
Instead of defining `embedded_url` (and all it's dependencies) on the class, I'd make them private instance methods of the tag class. While it's true that they don't rely on instance state, it's more in line with how our other tags do things.